### ISSUE TITLE
Targeting dom0 with pkg.installed containing setopt fails

### ISSUE THREAD

**Issue by 3isenHeiM** [ben-grande/qusal#100]
### Commitment

I confirm that I have read the following resources:

*   [How to troubleshoot Qusal](https://github.com/ben-grande/qusal/blob/main/docs/TROUBLESHOOT.md)
*   [How to ask questions The Smart Way](http://catb.org/esr/faqs/smart-questions.html)
*   [Writing the perfect question](https://codeblog.jonskeet.uk/2010/08/29/writing-the-perfect-question/)
*   [Question checklist](https://codeblog.jonskeet.uk/2012/11/24/stack-overflow-question-checklist/)
*   [Could you please make my preference the default?](https://www.qubes-os.org/faq/#could-you-please-make-my-preference-the-default)

### Software version
008d961821aa2f66996ce28fbb6972822b6bb785

### Brief summary

I'm trying to apply state files to install packages.

However, the dnf command resulting from the `pkg` state is erroring.

Here is the state file (salt/dom0/install-dev.sls):

```yaml
"{{ slsdotpath }}-dev-installed":
  pkg.installed:
    - require:
      - sls: utils.tools.common.update
    - install_recommends: False
    - skip_suggestions: True
    - setopt: "install_weak_deps=False"
    - pkgs:
      - vim
      - tmux
      - xclip
      - bash-completion
      - man-db
      - tree
```

The error I'm getting is 

```log
Setopt argument has no value: install
No such command: install_weak_deps=False. Please use /usr/bin/dnf --help
It could be a DNF plugin command, try: "dnf install 'dnf-command(install_weak_deps=False)'"
```
 And looking in the `/var/log/dnf.log` file, the issued command is 

```
dnf --exclude=qubes-template-* -y --best --allowerasing --setopt install install_weak_deps=False vim tmux [...]
```

### Steps to reproduce

Run the sls file above.

### Expected behavior

The dnf option `install_weak_deps=False ` should be set right after the `--setopt` flag.

### Actual behavior

The salt state is erroring with the errors mentionned above.

.

**Comment by ben-grande** [comment]
Thanks for reporting. I confirm it is broken when targetting dom0. Can you
try out targetting a fedora qube? I don't have this problem on domUs.

Can you try the sys-pgp.install on a fedora qube? Which Fedora version are
you using?

https://github.com/ben-grande/qusal/blob/main/salt/sys-pgp/install.sls

**Comment by ben-grande** [comment]
With the following example state (change the package scrot to a package you don't have installed already):

```salt
install:
  pkg.installed:
    - install_recommends: False
    - skip_suggestions: True
    - setopt: "install_weak_deps=False"
    - pkgs:
      - scrot
```

Targetting dom0:

```
DEBUG Loaded plugins: builddep, changelog, config-manager, copr, debug, debuginfo-install, download, generate_completion_cache, groups-manager, needs-restarting, playground, repoclosure, repodiff, repograph, repomanage, reposync, system-upgrade
DEBUG DNF version: 4.18.0
DDEBUG Command: dnf --exclude=qubes-template-* -y --best --allowerasing --setopt install install_weak_deps=False scrot
DDEBUG Installroot: /
DDEBUG Releasever: 4.2
DEBUG cachedir: /var/cache/dnf
CRITICAL No such command: install_weak_deps=False. Please use /usr/bin/dnf --help
CRITICAL It could be a DNF plugin command, try: "dnf install 'dnf-command(install_weak_deps=False)'"
```

Targetting a fedora-40 qube:

```
DEBUG Loaded plugins: builddep, changelog, config-manager, copr, debug, debuginfo-install, download, generate_completion_cache, groups-manager, needs-restarting, playground, qubes-hooks, repoclosure, repodiff, repograph, repomanage, reposync, system-upgrade
DEBUG DNF version: 4.21.1
DDEBUG Command: dnf -y --best --allowerasing --setopt install_weak_deps=False install scrot
DDEBUG Installroot: /
DDEBUG Releasever: 40
DEBUG cachedir: /var/cache/dnf
DDEBUG Base command: install
DDEBUG Extra commands: ['-y', '--best', '--allowerasing', '--setopt', 'install_weak_deps=False', 'install', 'scrot']
DEBUG User-Agent: constructed: 'libdnf (Fedora Linux 40; generic; Linux.x86_64)'
```

I don't know yet what can be the problem, maybe the DNF version or maybe the Qubes Dom0 Update mechanism.

**Comment by ben-grande** [comment]
- https://github.com/QubesOS/qubes-issues/issues/9509
- https://github.com/QubesOS/qubes-mgmt-salt-dom0-update/pull/30

**Comment by ben-grande** [comment]
Code was merged upstream, will be included in qubes-mgmt-salt-dom0-update version 4.3.2.
